tzdata dependency required for python>=3.9

I got this error on windows with python 3.10.11

```
..\build_venv\lib\site-packages\celery\bin\worker.py:13: in <module>
    from celery.concurrency.base import BasePool
..\build_venv\lib\site-packages\celery\concurrency\base.py:13: in <module>
    from celery.utils import timer2
..\build_venv\lib\site-packages\celery\utils\timer2.py:14: in <module>
    from kombu.asynchronous.timer import Entry
..\build_venv\lib\site-packages\kombu\asynchronous\__init__.py:7: in <module>
    from .hub import Hub, get_event_loop, set_event_loop
..\build_venv\lib\site-packages\kombu\asynchronous\hub.py:20: in <module>
    from .timer import Timer
..\build_venv\lib\site-packages\kombu\asynchronous\timer.py:32: in <module>
    EPOCH = datetime.fromtimestamp(0, ZoneInfo("UTC"))
C:\python\3.10\lib\zoneinfo\_common.py:24: in load_tzdata
    raise ZoneInfoNotFoundError(f"No time zone found with key {key}")
E   zoneinfo._common.ZoneInfoNotFoundError: 'No time zone found with key UTC'
```

Reproduce:

```
>>> from zoneinfo import ZoneInfo
>>> ZoneInfo("UTC")
Traceback (most recent call last):
  File "C:\python\3.10\lib\zoneinfo\_common.py", line 12, in load_tzdata
    return importlib.resources.open_binary(package_name, resource_name)
  File "C:\python\3.10\lib\importlib\resources.py", line 43, in open_binary
    package = _common.get_package(package)
  File "C:\python\3.10\lib\importlib\_common.py", line 66, in get_package
    resolved = resolve(package)
  File "C:\python\3.10\lib\importlib\_common.py", line 57, in resolve
    return cand if isinstance(cand, types.ModuleType) else importlib.import_module(cand)
  File "C:\python\3.10\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 992, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1004, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'tzdata'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\python\3.10\lib\zoneinfo\_common.py", line 24, in load_tzdata
    raise ZoneInfoNotFoundError(f"No time zone found with key {key}")
zoneinfo._common.ZoneInfoNotFoundError: 'No time zone found with key UTC'
```

This is solved by `pip install tzdata`.

The reason tzdata seems necessary: https://docs.python.org/3/library/zoneinfo.html#data-sources

```
The zoneinfo module does not directly provide time zone data, and instead pulls time zone
information from the system time zone database or the first-party PyPI package tzdata, if available.
Some systems, including notably Windows systems, do not have an IANA database available,
and so for projects targeting cross-platform compatibility that require time zone data, it is
recommended to declare a dependency on tzdata. If neither system data nor tzdata are
available, all calls to ZoneInfo will raise ZoneInfoNotFoundError.
```

Currently tzdata is only installed for python<3.9

https://github.com/celery/kombu/blob/33490adf9c52f2f003152827de8d0fc9f0e6c41a/requirements/default.txt#L4

--- a/requirements/default.txt
+++ b/requirements/default.txt
@@ -1,4 +1,5 @@
 typing_extensions==4.12.2; python_version<"3.10"
 amqp>=5.1.1,<6.0.0
 vine==5.1.0
-backports.zoneinfo[tzdata]>=0.2.1; python_version < '3.9'
+backports.zoneinfo[tzdata]>=0.2.1; python_version<"3.9"
+tzdata; python_version>="3.9"

